lib/elixir/pages/anti-patterns/code-anti-patterns.md

```diff
@@ -0,0 +1,39 @@
+# Code-related anti-patterns
+
+This document outlines anti-patterns related to your code and particular Elixir idioms and features.
+
+## Comments
+
+TODO.
+
+## Long parameter list
+
+TODO.
+
+## Complex branching
+
+TODO.
+
+## Complex else clauses in with
+
+TODO.
+
+## Complex extractions in clauses
+
+TODO.
+
+## Dynamic map fields access
+
+TODO.
+
+## Dynamic atom creation
+
+TODO.
+
+## Namespace violation
+
+TODO. (note: this is defining modules not in your namespace)
+
+## Speculative Generality
```

Comment:
I know this is a TODO but we might want to use the right casing that we use in Elixir

```suggestion
## Speculative generality
```